src/UDB2/Media/ImageCollectionFactory.php

```diff
@@ -57,17 +57,25 @@ public function withUuidRegex($mediaIdentifierRegex)
      */
     public function fromUdb2Item(CultureFeed_Cdb_Item_Base $item)
     {
-        $dutch = new Language('nl');
-        $dutchDetail = $item
-            ->getDetails()
-            ->getDetailByLanguage('nl');
-        $title = $dutchDetail->getTitle();
+        $details = $item->getDetails();
+        $detail = $details->getDetailByLanguage('nl');
+
+        if (!$detail) {
+            $details->rewind();
+            $detail = $details->current();
```

Comment:
This will break the expected behaviour and unit tests because it will always return an empty `ImageCollection` if no `nl` detail is found if you put the early return in the first check.